### ISSUE TITLE
Multitask Training for Similarity and Classification

### ISSUE THREAD

**Issue by rg089** [UKPLab/sentence-transformers#1341]
@nreimers Thanks for the amazing library!

I had a question about multitask training for similarity detection and single sentence classification. Basically, the first task is just detecting the similarity between a pair of sentences given as inputs (using any of the existing losses). The second task is training both the sentences in the pair separately for classification.

What would be the best way to achieve this using the library? Should a separate loss function be defined similar to the softmax loss which classifies a single input instead of a pair? 

Thanks!

**Comment by nreimers** [comment]
Yes, a new loss makes sense

**Comment by rg089** [comment]
@nreimers Thanks for the reply!

Just wanted to confirm one thing: Since the dataset is the same for both the tasks, and classification is a sort of auxiliary supporting task, should I create a loss function that incorporates both the softmax loss and contrastive loss?

Or would training it in the same way as the multitask script in the examples, where we define two separate losses, make more sense?

Thanks again!

**Comment by nreimers** [comment]
If you have always both labels for both task, you can do it in one loss. If not, i.e. if the datasets for the two tasks are different, you need two different losses
